read_write.go

```diff
@@ -0,0 +1,95 @@
+package epplib
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"math"
+
+	"github.com/pkg/errors"
+)
+
+// ErrMessageSize represent an error where the incoming message size is bigger than allowed.
+var ErrMessageSize = errors.New("message size exceeds limit")
```

Comment:
Perhaps we can use stdlib here?

Reply:
"github.com/pkg/errors" is needed for errors.Wrapf on line 33, but that could be updated if you think we should switch to the stdlib?

Reply:
Sure but the `ErrMessageSize` root doesn't need to be from `pkg/errors`.